src/website/shared/evaluation.py

```diff
@@ -0,0 +1,348 @@
+import asyncio
+import json
+from dataclasses import dataclass, field
+from typing import Any, Generator
+
+from asgiref.sync import sync_to_async
+from dataclass_wizard import DumpMixin, JSONWizard, LoadMixin
+
+from shared.models.nix_evaluation import (
+    NixDerivation,
+    NixDerivationMeta,
+    NixDerivationOutput,
+    NixEvaluation,
+    NixLicense,
+    NixMaintainer,
+    NixOutput,
+    NixPlatform,
+    NixStorePathOutput,
+)
+
+
+@dataclass
+class MaintainerAttribute(JSONWizard):
+    name: str
+    github: str | None = None
+    githubId: int | None = None
+    email: str | None = None
+    matrix: str | None = None
+
+
+@dataclass
+class LicenseAttribute(JSONWizard):
+    fullName: str | None = None
+    deprecated: bool = False
+    free: bool = False
+    redistributable: bool = False
+    shortName: str | None = None
+    spdxId: str | None = None
+    url: str | None = None
+
+
+@dataclass
+class MetadataAttribute(JSONWizard, LoadMixin, DumpMixin):
+    outputsToInstall: list[str] = field(default_factory=list)
+    available: bool = True
+    broken: bool = False
+    unfree: bool = False
+    unsupported: bool = False
+    insecure: bool = False
+    mainProgram: str | None = None
+    position: str | None = None
+    homepage: str | None = None
+    description: str | None = None
+    name: str | None = None
+    maintainers: list[MaintainerAttribute] = field(default_factory=list)
+    license: list[LicenseAttribute] = field(default_factory=list)
+    platforms: list[str] = field(default_factory=list)
+    knownVulnerabilities: list[str] = field(default_factory=list)
+
+    def __pre_as_dict__(self):
+        linearized_maintainers = []
+        for maintainer in self.maintainers:
+            if maintainer.get("scope") is not None:  # pyright: ignore generalTypeIssue
+                linearized_maintainers.extend(
+                    maintainer.get("members", [])  # pyright: ignore generalTypeIssue
+                )
+            else:
+                linearized_maintainers.append(maintainer)
+        self.maintainers = linearized_maintainers
+
+
+@dataclass
+class EvaluatedAttribute(JSONWizard):
+    """
+    This is a totally evaluated attribute.
+    """
+
+    attr: str
+    attrPath: list[str]
+    name: str
+    drvPath: str
```

Comment:
drv_path

Reply:
yeah all of this needs rebase over your formatting changes

Reply:
Maybe smaller PRs would help to avoid this?

Reply:
https://github.com/tummychow/git-absorb should help

Reply:
Right but @Mic92 your reformatting actually broke I believe things. So I am glad I didn't do smaller PRs :P